src/global/helpers/arrow-navigation.ts

```diff
@@ -0,0 +1,54 @@
+import getDocumentWritingMode from './get-document-writing-mode';
+
+/**
+ * Check if the key pressed is among those allowed for navigation.
+ * @param event The keyboard event to check.
+ */
+export function isArrowKeyPressed(event: KeyboardEvent): boolean {
+  return ['ArrowRight', 'ArrowLeft', 'ArrowUp', 'ArrowDown'].includes(event.key);
+}
+
+/**
+ * Checks the direction of the navigation event in the current writing mode.
+ * Returns the right function for calculate the next element's index,
+ * based on the of the current element's index and the size of the list to navigate.
+ * @param event The event to check.
+ * @returns A function to calculate the index of the element to move to:
+ * if it is a 'previous' event, returns a function to get the index of the previous element,
+ * or, if the current is the first in the list, the index of the last one;
+ * if it is a 'next' event, returns a function to get the index of the next element,
+ * or, if the current is the last in the list, the index of the first one.
+ */
+function getNextElementIndexFunction(
+  event: KeyboardEvent
+): (currentElementIndex: number, size: number) => number {
+  let prevKey: string, nextKey: string;
+  if (getDocumentWritingMode() === 'rtl') {
+    prevKey = 'ArrowRight';
+    nextKey = 'ArrowLeft';
+  } else {
+    prevKey = 'ArrowLeft';
+    nextKey = 'ArrowRight';
+  }
+
+  if (event.key === prevKey || event.key === 'ArrowUp') {
+    return (currentElementIndex: number, size: number) =>
```

Comment:
I'm not sure about returning functions that will in turn return the index; is there a particular reason for this? Otherwise, it could be simplified by returning the index directly and perhaps using the math module as Jeri suggested:

```ts
* @returns The index of the element to be selected: 
 * if it is a "previous" event, the index of the previous element, 
 * or the index of the last element if the current one is the first in the list; 
 * if it is a "next" event, the index of the next element, 
 * or the index of the first element if the current one is the last in the list.
 */
export function getNextElementIndex(
  event: KeyboardEvent,
  current: number, 
  size: number
): number {
  let prevKey: string, nextKey: string;
  if (getDocumentWritingMode() === 'rtl') {
    prevKey = 'ArrowRight';
    nextKey = 'ArrowLeft';
  } else {
    prevKey = 'ArrowLeft';
    nextKey = 'ArrowRight';
  }

  if (event.key === prevKey || event.key === 'ArrowUp') {
    return (current - 1 + size) % size;
  } else if (event.key === nextKey || event.key === 'ArrowDown') {
    return (current + 1 + size) % size;
  }
}
```

Reply:
In a previous implementation I didn't pass the "current" and "size" params, but now I do, so it's ok to fix as suggested 👍